File: ios/Classes/Models/HMSRemovedFromRoomExtension.swift
//
//  HMSRemovedFromRoomExtension.swift
//  hmssdk_flutter
//
//  Created by Yogesh Singh on 06/01/22.
//

import Foundation
import HMSSDK

class HMSRemovedFromRoomExtension {

    static func toDictionary(_ notification: HMSRemovedFromRoomNotification) -> [String: Any] {

        var dict = [String: Any]()

        dict["reason"] = notification.reason

        dict["room_was_ended"] = notification.roomEnded

        if let peer = notification.requestedBy {
            dict["peer_who_removed"] = HMSPeerExtension.toDictionary(peer)
        }

        return dict
    }
}

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HMSRemovedFromRoomExtension.kt
package live.hms.hmssdk_flutter

import live.hms.video.sdk.models.HMSRemovedFromRoom

class HMSRemovedFromRoomExtension {
    companion object{
        fun toDictionary(notification:HMSRemovedFromRoom?): HashMap<String, Any?>? {
            val hashMap = HashMap<String,Any?>()

            if(notification==null)return null
            hashMap.put("peer_who_removed",HMSPeerExtension.toDictionary(notification.peerWhoRemoved))
            hashMap.put("reason",notification.reason)
            hashMap.put("room_was_ended",notification.roomWasEnded)

            val roomMap = HashMap<String,Any?>()
            roomMap.put("removed_from_room",hashMap)
            return roomMap
        }
    }
}